### ISSUE TITLE
Statusbar takes seconds to update

### ISSUE THREAD

**Issue by mallet** [awesomeWM/awesome#925]
Hi,

When a new application is launched, when I select a new window, when I switch to another tag, etc. the statusbar is not updated immediately. Actually it seems to be updated when the clock is updated. I can have the statusbar updated every 0.1s by forcing such refresh rate for the clock. But then, awesome constantly takes 10% of the CPU.

This happens with awesome 3.5.6 as available under Ubuntu 16.04. The problem manifests itself only when the screen is rotated in "portrait" mode, not when it is in the classical "landscape" mode. Previously, I think I used awesome 3.4.8 on the same computer, without any issues. And before you ask, I properly updated "rc.lua", starting from the one found in "/etc/xdg/awesome/".

Thanks for the great work!

**Comment by Elv13** [comment]
please try starting awesome with `--no-argb`, this something helps.

If this doesn't please try the current awesome git master branch. You can generate Ubuntu packages using `make package` once Awesome is compiled. The dependencies are the same as Awesome 3.5.6, so it should work fine.

**Comment by mallet** [comment]
Using "--no-argb" apparently solved it completely!

(As a side effect, the font size in a new instance of Emacs is now correct in portrait mode, while it was huge previously.)

Thank you for your help!

**Comment by Elv13** [comment]
No problem. This happen because of a buggy graphic driver. We don't intend to figure those ones anymore.

**Comment by psychon** [comment]
Still, feel free to report a bug upstream against your graphic drivers. If you do so, leave a link here and I might fill in some technical details. (By default awesome uses the 32 bit visual, `--no-argb` makes it use the server's default visual; I don't know why tasklist updates do something different from clock updates. To the X11 server, both should look the same, so I guess the only difference here is "a new window appeared elsewhere")

If you don't do so, please still tell us which graphic drivers you use so that we have something like an overview of the issues.

**Comment by mallet** [comment]
Hello,

On Fri, May 27 2016, Uli Schlachter wrote:
> Still, feel free to report a bug upstream against your graphic drivers. If you do so, leave a link here and I might fill in some technical details. (By default awesome uses the 32 bit visual, `--no-argb` makes it use the server's default visual; I don't know why tasklist updates do something different from clock updates. To the X11 server, both should look the same, so I guess the only difference here is "a new window appeared elsewhere")
>
> If you don't do so, please still tell us which graphic drivers you use so that we have something like an overview of the issues.

Here is what "lshw -C display" says:

  *-display               
       description: VGA compatible controller
       product: Caicos [Radeon HD 6450/7450/8450 / R5 230 OEM]
       vendor: Advanced Micro Devices, Inc. [AMD/ATI]
       physical id: 0
       bus info: pci@0000:01:00.0
       version: 00
       width: 64 bits
       clock: 33MHz
       capabilities: pm pciexpress msi vga_controller bus_master cap_list rom
       configuration: driver=radeon latency=0
       resources: irq:28 memory:e0000000-efffffff memory:f7e20000-f7e3ffff ioport:e000(size=256) memory:f7e00000-f7e1ffff

So, the driver is « radeon ».

Best,
VM.

**Comment by Elv13** [comment]
I never had a 6450, but I did have a 5450 at some point. I can tell you this. Those cards are buggggggy. The driver doesn't make this any better. In my case, there was an hidden (resolution) buffer size limitation, if exceeded, garbage would be displayed. It was also showing 1 big screen in xrandr even when xorg.conf was configured with 3 screens. I also had this argb issue, but by then I gave up on that card. I still use it in my motherboard testing rig because it is fan-less and rock solid, but I don't recommend the *450 serie as a desktop card, even for an extra screen. Better get an older NVIDIA GTX x50 or x60 card. They are also available in single slot versions.
